src/functionals.jl

```diff
@@ -22,6 +22,13 @@ function expectation(distr::UnivariateDistribution, g::Function)
     expectation(distr, g, 1e-10)
 end
 
+function expectation(distr::MultivariateDistribution, g::Function; nsamples::Int=100, rng::AbstractRNG=GLOBAL_RNG)
+    nsamples > 0 || throw(ArgumentError("number of samples should be > 0"))
```

Comment:
I think it's helpful to add a comment here that explains why a separate function is used:
```suggestion
    nsamples > 0 || throw(ArgumentError("number of samples should be > 0"))
    # We use a function barrier to work around type instability of `sampler(dist)`
```